apps/studio/src/server/modules/collection/collection.router.ts

```diff
@@ -349,4 +350,21 @@ export const collectionRouter = router({
         })
       },
     ),
+  getCollections: protectedProcedure
+    .input(getCollectionsSchema)
+    .query(async ({ ctx, input: { siteId } }) => {
+      await validateUserPermissionsForResource({
+        siteId,
+        action: "read",
+        userId: ctx.user.id,
+      })
```

Comment:
issue: hmm I think it should be `validateUserPermissionsForSite`? Cos will be querying over the entire site's collections